### PR TITLE
Subjects

### PR DESCRIPTION
**Pull request by leduytungframgia**

### CODE REVIEW

File: app/views/subjects/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<% provide :title, 'Edit Subject' %>
+<%= form_for @subject do |f| %>
+  <%= render 'shared/error_messages', object: f.object %>
+  <div class="field">
+    <%= f.text_field :name %>
+  </div>
+  <div class="field">
+    <%= f.text_area :instruction%>
```
Comment (NguyenTanDuc) [inline app/views/subjects/edit.html.erb]: add space in %>

File: app/views/subjects/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<% provide :title, @subject.name %>
+<div class="panel panel-info">
+    <div class="alert alert-info" role="alert"><span class="user">
+      <h3><%= link_to @subject.name, @subject %></h3>
+      <span class="timestamp">
+      Posted <%= time_ago_in_words @subject.created_at %> ago.
+
+      <%= link_to "Delete", @entry, method: :delete,
+                                    data: {confirm: "You sure?"} %>
```
Comment (NguyenTanDuc) [inline app/views/subjects/show.html.erb]: I18n "You sure?"

File: app/controllers/subjects_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+class SubjectsController < ApplicationController
```
Comment (chuanhtuan) [inline app/controllers/subjects_controller.rb]: admin namespace

File: config/routes.rb [hunk 4]
```diff
@@ -1,4 +1,6 @@
 Rails.application.routes.draw do
   root             'static_pages#home'
   get 'help'    => 'static_pages#help'
+
+  resources :subjects
```
Comment (chuanhtuan) [inline config/routes.rb]: admin namespace

File: app/controllers/supervisor/subjects_controller.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+class Supervisor::SubjectsController < ApplicationController
+  def index
+    @subjects = Subject.paginate page: params[:page], per_page: 20
+  end
```
Comment (NguyenTanDuc) [inline app/controllers/supervisor/subjects_controller.rb]: Add new line

File: app/views/subjects/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<% provide :title, @subject.name %>
+<div class="panel panel-info">
+    <div class="alert alert-info" role="alert"><span class="user">
+      <h3><%= link_to @subject.name, @subject %></h3>
+      <span class="timestamp">
+      Posted <%= time_ago_in_words @subject.created_at %> ago.
+
```
Comment (NguyenTanDuc) [inline app/views/subjects/show.html.erb]: remove this line

File: app/controllers/subjects_controller.rb (hunk 3 above)
Comment (leduytungframgia) [inline app/controllers/subjects_controller.rb]: supervisor namespace added

File: app/controllers/supervisor/subjects_controller.rb (hunk 5 above)
Comment (leduytungframgia) [inline app/controllers/supervisor/subjects_controller.rb]: added

**Pull request merged**